File: apps/web/src/types/device-package.ts
export type PackageActivationStatus = 'pending' | 'succeeded' | 'rejected' | 'failed'

export interface PackageActivation {
  id: number
  package_slot: number
  package_code_suffix: string
  device_id: number
  device_name: string
  device_serial_masked: string
  channel_no: number
  activation_status: PackageActivationStatus
  official_code: string | null
  official_message: string | null
  activated_at: string | null
  activated_by: number
  retry_count: number
  created_at: string
  updated_at: string
}

export interface PackageSlot {
  slot: number
  configured: boolean
  activation: PackageActivation | null
}

export type PackageEntitlementBlocker =
  | 'ezviz_credentials_not_configured'
  | 'token_not_authenticated'
  | 'no_package_codes_configured'
  | 'no_online_devices'

export interface PackageEntitlementSummary {
  source: 'competition_notice'
  package_slots_total: number
  configured_slot_count: number
  activated_slot_count: number
  validity_months: number
  coupon_redeemed: boolean
  token_status: 'not_configured' | 'not_cached' | 'valid' | 'refresh_required'
  online_device_count: number
  activation_ready: boolean
  blockers: PackageEntitlementBlocker[]
}

export interface PackageActivationRequest {
  package_slot: number
  device_id: number
  channel_no: number
  confirmed: true
}

File: services/api/app/schemas/device_package.py
from datetime import datetime
from typing import Literal

from pydantic import BaseModel, Field

from app.models.device_package import PackageActivationStatus


class PackageActivationRead(BaseModel):
    id: int
    package_slot: int
    package_code_suffix: str
    device_id: int
    device_name: str
    device_serial_masked: str
    channel_no: int
    activation_status: PackageActivationStatus
    official_code: str | None
    official_message: str | None
    activated_at: datetime | None
    activated_by: int
    retry_count: int
    created_at: datetime
    updated_at: datetime


class PackageSlotRead(BaseModel):
    slot: int
    configured: bool
    activation: PackageActivationRead | None


class PackageEntitlementSummary(BaseModel):
    source: Literal["competition_notice"] = "competition_notice"
    package_slots_total: int = 5
    configured_slot_count: int
    activated_slot_count: int
    validity_months: int = 6
    coupon_redeemed: bool
    token_status: str
    online_device_count: int
    activation_ready: bool
    blockers: list[str]


class PackageActivationRequest(BaseModel):
    package_slot: int = Field(ge=1, le=5)
    device_id: int = Field(gt=0)
    channel_no: int = Field(gt=0)
    confirmed: Literal[True]
